### PR TITLE
[Doc] Add promptflow model flavor documentation

### PR DESCRIPTION
**Pull request by brynn-code**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/brynn-code/mlflow/pull/11385?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11385/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11385
```

</p>
</details>

### Related Issues/PRs

#11311 

Preview the docs [here](https://output.circle-artifacts.com/output/job/179d9d79-7ff6-462c-98cf-24ff2ad09b12/artifacts/0/docs/build/html/models.html#promptflow-promptflow-experimental)

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [x] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [x] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by brynn-code** [comment]
@B-Step62 Please help review the doc changes when you have time, thanks in advance! :)

**Comment by brynn-code** [comment]
@serena-ruan @harupy Could you please help take a look at the doc changes when you have time? Thanks!

**Review by B-Step62 — APPROVED** [review]
LGTM, thanks for working on this super quick!

**Comment by brynn-code** [comment]
I've reviewed the doc with our team members, please help to merge it anytime :)

File: docs/source/models.rst [hunk 1]
```diff
@@ -2128,6 +2128,30 @@ Example:
...
+:py:func:`mlflow.pyfunc.load_model()`. You can also use the :py:func:`mlflow.promptflow.load_model()`
+method to load MLflow Models with the ``promptflow`` model flavor in native promptflow format.
+
+Please note that the ``signature`` in ``MLmodel`` file will NOT BE automatically inferred from the flow itself.
+To save model with the signature, you can either pass the ``input_example`` or specify the input signature manually.
+
+
+Example:
+
+.. literalinclude:: ../../examples/promptflow/train.py
```
Comment (serena-ruan) [inline docs/source/models.rst:2151]: When this displays on the website, the pure file content is not executable directly. Could we add a comment above `flow_folder = Path(__file__).parent / "basic"` to let users know where this file lives inside mlflow repo? Or a comment above the code section should be helpful :)

Comment (brynn-code) [reply docs/source/models.rst:2151]: Okay, good point!

**Pull request merged**